### PR TITLE
[fallible][easy] copy_from_slice -> copy_slice_to_vec

### PR DESCRIPTION
**Pull request by mimoo**
making use of the fallible `copy_slice_to_vec` wherever possible.

### CODE REVIEW

File: crypto/crypto/Cargo.toml [hunk 1]
```diff
@@ -36,10 +36,12 @@ tiny-keccak = { version = "2.0.2", features = ["sha3"] }
 vanilla-x25519-dalek = { version = "1.1.0", package = 'x25519-dalek', default-features = false, features = ["std"], optional = true }
 x25519-dalek = { git = "https://github.com/novifinancial/x25519-dalek.git", branch = "fiat5", version = "1.1.0", default-features = false, features = ["std", "fiat_u64_backend"], optional = true}
 aes-gcm = "0.8.0"
-diem-crypto-derive = { path = "../crypto-derive", version = "0.1.0" }
+
 bcs = "0.1.2"
+diem-crypto-derive = { path = "../crypto-derive", version = "0.1.0" }
 diem-nibble = { path = "../../common/nibble", version = "0.1.0" }
 diem-workspace-hack = { path = "../../common/workspace-hack", version = "0.1.0" }
+fallible = { path = "../../common/fallible", version = "0.1.0" }
```
Comment (bmwill) [inline crypto/crypto/Cargo.toml:44]: So one thing i'm working on is being able to publish the crypto crate and I would really like if we could avoid adding *another* dep to this list that needs to be considered published to crates.io.

Comment (mimoo) [reply crypto/crypto/Cargo.toml:44]: ah interesting, is fallible the only intruder here? Also what's the reason for publishing the crypto crate : o ? It currently mostly contains wrappers

Comment (kchalkias) [reply crypto/crypto/Cargo.toml:44]: It's not only a wrapper? But it's true that many of our updates re key security have been accepted from Dalek authors to upstream, a few of them re malleability not yet though.

Comment (mimoo) [reply crypto/crypto/Cargo.toml:44]: good point! I'm wondering if these will eventually make it to our fork of dalek :P

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A fallible 0.1.0 (workspace, path 'common/fallible')
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A fallible 0.1.0 (workspace, path 'common/fallible')
    * features: [none]
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A fallible 0.1.0 (workspace, path 'common/fallible')
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A fallible 0.1.0 (workspace, path 'common/fallible')
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A fallible 0.1.0 (workspace, path 'common/fallible')
    * features: [none]
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A fallible 0.1.0 (workspace, path 'common/fallible')
    * features: [none]
```

**Comment by mimoo** [comment]
I'm getting a lot of 

> Error:  [unpublished-packages-only-use-path-dependencies] [package 'network' (at network)]: unpublished package specifies a version of first-party dependency 'fallible'; unpublished packages should only use path dependencies for first-party packages.

which makes me think that we'd probably want to work on making common/fallible and common/infallible published first

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A fallible 0.1.0 (workspace, path 'common/fallible')
    * features: [none]
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A fallible 0.1.0 (workspace, path 'common/fallible')
    * features: [none]
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A fallible 0.1.0 (workspace, path 'common/fallible')
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A fallible 0.1.0 (workspace, path 'common/fallible')
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A fallible 0.1.0 (workspace, path 'common/fallible')
    * features: [none]
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A fallible 0.1.0 (workspace, path 'common/fallible')
    * features: [none]
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A fallible 0.1.0 (workspace, path 'common/fallible')
    * features: [none]
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A fallible 0.1.0 (workspace, path 'common/fallible')
    * features: [none]
```

**Comment by davidiw** [comment]
are we going to wrap this up :O

**Pull request closed without merging**